From f5edd93b52e9ebf17fc4b8776f016914ba4bad49 Mon Sep 17 00:00:00 2001
From: Guillaume Legoupil <guillaume.legoupil@nxp.com>
Date: Fri, 6 Mar 2026 09:50:58 +0100
Subject: TMP Fix MCUBoot & OTA: don't shutdown NBU


diff --git a/soc/nxp/mcx/mcxw/mcxw7xx/soc.c b/soc/nxp/mcx/mcxw/mcxw7xx/soc.c
index d74714ee2dc..9160bc5c6dd 100644
--- a/soc/nxp/mcx/mcxw/mcxw7xx/soc.c
+++ b/soc/nxp/mcx/mcxw/mcxw7xx/soc.c
@@ -318,6 +318,8 @@ static int soc_nbu_init(void)
 {
 #if defined(CONFIG_NXP_NBU)
 	nxp_nbu_init();
+#elif defined(CONFIG_MCUBOOT)
+	/* Keep the NBU as is, wait for MCUBOOT's Image to configure it */
 #else
 	/* Shutdown NBU as not used */
 
