From 97b56f4c240d35212383ac12578e6e80c16e753f Mon Sep 17 00:00:00 2001
From: Guillaume Legoupil <guillaume.legoupil@nxp.com>
Date: Fri, 31 Oct 2025 08:56:13 +0100
Subject: mcux: middleware: ieee_802_15_4: add debug rpmsg trace


diff --git a/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/mac/source/App/mac_intf_rpmsg.c b/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/mac/source/App/mac_intf_rpmsg.c
index 39e624994..511d6ccae 100644
--- a/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/mac/source/App/mac_intf_rpmsg.c
+++ b/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/mac/source/App/mac_intf_rpmsg.c
@@ -76,6 +76,8 @@ extern const uint8_t gUseRtos_c;
 
 uint16_t mlmeGetSizeOfPIB(pibId_t pib);
 
+void wcs_trace_rpmsg_mac(bool tx, macMessageId_t msgType, void *pMsg, instanceId_t macInstanceId, resultType_t ret);
+
 static void mac_msg_free(nwkMessage_t *msg)
 {
     if ((msg->msgType == gMlmeScanCnf_c) && (msg->msgData.scanCnf.scanType == gScanModeED_c))
@@ -96,6 +98,7 @@ static void mac_intf_cb_task_func()
         {
             msg_free = TRUE;
 
+            wcs_trace_rpmsg_mac(false, msg->msgType, msg, msg->mac_id, gSuccess_c);
             switch (msg->msgType)
             {
                 case gMcpsDataInd_c:
@@ -350,6 +353,7 @@ resultType_t NWK_MLME_SapHandler(mlmeMessage_t* req, nwkMessage_t *cnf, instance
 
     if (!mac_intf_init_done)
     {
+        wcs_trace_rpmsg_mac(true, req->msgType, req, macInstanceId, gInvalidParameter_c);
         return gInvalidParameter_c;
     }
 
@@ -395,17 +399,32 @@ resultType_t NWK_MLME_SapHandler(mlmeMessage_t* req, nwkMessage_t *cnf, instance
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)mac_rpmsg_handle, msg, len) != kStatus_HAL_RpmsgSuccess)
         {
+            wcs_trace_rpmsg_mac(true, req->msgType, req, macInstanceId, gInvalidParameter_c);
             ret = gInvalidParameter_c;
             break;
         }
+        wcs_trace_rpmsg_mac(true, req->msgType, req, macInstanceId, ret);
 
         if (req->msgType == gMlmeGetReq_c)
         {
             wait_response();
             wait_mac_rsp = FALSE;
 
+            cnf->msgData.getCnf.pibAttribute = req->msgData.getReq.pibAttribute;
             cnf->msgData.getCnf.pibAttributeLength = mac_get_pib_rsp.pibAttributeLength;
             cnf->msgData.getCnf.pibAttributeValue = mac_get_pib_rsp.pibAttributeValue;
+            wcs_trace_rpmsg_mac(false, req->msgType, cnf, macInstanceId, ret);
+        }
+        if (req->msgType == gMlmeStartReq_c)
+        {
+            uint8_t      msgType   = cnf->msgType;
+            resultType_t msgStatus = cnf->msgData.startCnf.status;
+
+            cnf->msgType                 = gMlmeStartCnf_c;
+            cnf->msgData.startCnf.status = ret;
+            wcs_trace_rpmsg_mac(false, req->msgType, req, macInstanceId, ret);
+            cnf->msgType                 = msgType;
+            cnf->msgData.startCnf.status = msgStatus;
         }
     } while (0);
 
@@ -432,6 +451,7 @@ resultType_t NWK_MCPS_SapHandler(nwkToMcpsMessage_t* pMsg, instanceId_t macInsta
 
     if (!mac_intf_init_done)
     {
+        wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, gInvalidParameter_c);
         return gInvalidParameter_c;
     }
 
@@ -454,9 +474,11 @@ resultType_t NWK_MCPS_SapHandler(nwkToMcpsMessage_t* pMsg, instanceId_t macInsta
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)mac_rpmsg_handle, (uint8_t *)pMsg, len) != kStatus_HAL_RpmsgSuccess)
         {
+            wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, gInvalidParameter_c);
             ret = gInvalidParameter_c;
             break;
         }
+        wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, ret);
     } while (0);
 
     PLATFORM_RemoteActiveRel();
@@ -570,6 +592,7 @@ resultType_t MAC_Proxy_PLME_SapHandler(macToPlmeMessage_t * pMsg, instanceId_t m
 
     if (!mac_intf_init_done)
     {
+        wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, gInvalidParameter_c);
         return gInvalidParameter_c;
     }
 
@@ -595,16 +618,20 @@ resultType_t MAC_Proxy_PLME_SapHandler(macToPlmeMessage_t * pMsg, instanceId_t m
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)mac_rpmsg_handle, (uint8_t *)pMsg, sizeof(macToPlmeMessage_t)) != kStatus_HAL_RpmsgSuccess)
         {
+            wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, gInvalidParameter_c);
             ret = gInvalidParameter_c;
             break;
         }
+        wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, ret);
 
         if (pMsg->msgType == gPlmeGetReq_c)
         {
             wait_response();
             wait_proxy_rsp = FALSE;
 
+            pMsg->msgData.getReq.PibAttribute = phy_get_rsp.PibAttribute;
             pMsg->msgData.getReq.PibAttributeValue = phy_get_rsp.PibAttributeValue;
+            wcs_trace_rpmsg_mac(false, pMsg->msgType, pMsg, macInstanceId, ret);
         }
     } while (0);
 
@@ -622,6 +649,7 @@ resultType_t MAC_Plat_Cmd(macPlatMessage_t *pMsg, instanceId_t macInstanceId)
 
     if (!mac_intf_init_done)
     {
+        wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, gInvalidParameter_c);
         return gInvalidParameter_c;
     }
 
@@ -647,9 +675,11 @@ resultType_t MAC_Plat_Cmd(macPlatMessage_t *pMsg, instanceId_t macInstanceId)
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)mac_rpmsg_handle, (uint8_t *)pMsg, sizeof(macPlatMessage_t) + pMsg->data_len) != kStatus_HAL_RpmsgSuccess)
         {
+            wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, gInvalidParameter_c);
             ret = gInvalidParameter_c;
             break;
         }
+        wcs_trace_rpmsg_mac(true, pMsg->msgType, pMsg, macInstanceId, ret);
 
         if (pMsg->is_get)
         {
@@ -666,6 +696,7 @@ resultType_t MAC_Plat_Cmd(macPlatMessage_t *pMsg, instanceId_t macInstanceId)
                 memcpy((uint8_t *)pMsg + sizeof(macPlatMessage_t), (uint8_t *)plat_get_rsp + sizeof(macPlatMessage_t), pMsg->data_len);
 
                 MSG_Free(plat_get_rsp);
+                wcs_trace_rpmsg_mac(false, pMsg->msgType, pMsg, macInstanceId, ret);
             }
         }
     } while (0);
diff --git a/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/phy/source/SerialDevice/Phy.c b/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/phy/source/SerialDevice/Phy.c
index 5dc084074..8dcd4172b 100644
--- a/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/phy/source/SerialDevice/Phy.c
+++ b/mcux/middleware/mcux-sdk-middleware-ieee_802.15.4/ieee_802_15_4/phy/source/SerialDevice/Phy.c
@@ -74,6 +74,9 @@ static bool_t phy_intf_init_ongoing = FALSE;
 extern const uint8_t gUseRtos_c;
 
 
+void wcs_trace_rpmsg_phy(bool tx, phyMessageId_t msgType, macToPlmeMessage_t *pMsg, instanceId_t phyInstance, phyStatus_t ret);
+
+
 static void wait_response()
 {
     osa_event_flags_t flags;
@@ -118,6 +121,7 @@ static void phy_intf_cb_task_func()
                     msg->ctx_id = 0;
                 }
 
+                wcs_trace_rpmsg_phy(false, gPhyExtCmd, (macToPlmeMessage_t *)msg, msg->ctx_id, gPhySuccess_c);
                 if (phyLocal[msg->ctx_id].ext_cmd_handler)
                 {
                     ext_phy_cmd_t *m = (ext_phy_cmd_t *)msg;
@@ -131,6 +135,7 @@ static void phy_intf_cb_task_func()
             }
             else
             {
+                wcs_trace_rpmsg_phy(false, msg->msgType, (macToPlmeMessage_t *)msg, msg->ctx_id, gPhySuccess_c);
                 switch (msg->msgType)
                 {
                     case gPdDataInd_c:
@@ -312,6 +317,7 @@ phyStatus_t MAC_PLME_SapHandler(macToPlmeMessage_t *pMsg, instanceId_t phyInstan
 
     if (!phy_intf_init_done)
     {
+        wcs_trace_rpmsg_phy(true, pMsg->msgType, pMsg, phyInstance, gPhyInvalidParameter_c);
         return gPhyInvalidParameter_c;
     }
 
@@ -340,9 +346,11 @@ phyStatus_t MAC_PLME_SapHandler(macToPlmeMessage_t *pMsg, instanceId_t phyInstan
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)phyRpmsgHandle, (uint8_t *)pMsg, sizeof(macToPlmeMessage_t)) != kStatus_HAL_RpmsgSuccess)
         {
+            wcs_trace_rpmsg_phy(true, pMsg->msgType, pMsg, phyInstance, gPhyBusy_c);
             ret = gPhyBusy_c;
             break;
         }
+        wcs_trace_rpmsg_phy(true, pMsg->msgType, pMsg, phyInstance, ret);
 
         if (pMsg->msgType == gPlmeGetReq_c)
         {
@@ -350,6 +358,7 @@ phyStatus_t MAC_PLME_SapHandler(macToPlmeMessage_t *pMsg, instanceId_t phyInstan
             wait_phy_rsp = FALSE;
 
             pMsg->msgData.getReq.PibAttributeValue = phy_get_rsp.PibAttributeValue;
+            wcs_trace_rpmsg_phy(false, pMsg->msgType, pMsg, phyInstance, ret);
         }
     } while(0);
 
@@ -376,6 +385,7 @@ phyStatus_t MAC_PD_SapHandler(macToPdDataMessage_t *pMsg, instanceId_t phyInstan
 
     if (!phy_intf_init_done)
     {
+        wcs_trace_rpmsg_phy(true, pMsg->msgType, (macToPlmeMessage_t *)pMsg, phyInstance, gPhyInvalidParameter_c);
         return gPhyInvalidParameter_c;
     }
 
@@ -398,9 +408,11 @@ phyStatus_t MAC_PD_SapHandler(macToPdDataMessage_t *pMsg, instanceId_t phyInstan
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)phyRpmsgHandle, (uint8_t *)pMsg, len) != kStatus_HAL_RpmsgSuccess)
         {
+             wcs_trace_rpmsg_phy(true, pMsg->msgType, (macToPlmeMessage_t *)pMsg, phyInstance, gPhyBusy_c);
              ret = gPhyBusy_c;
              break;
         }
+        wcs_trace_rpmsg_phy(true, pMsg->msgType, (macToPlmeMessage_t *)pMsg, phyInstance, ret);
     } while (0);
 
     PLATFORM_RemoteActiveRel();
@@ -457,6 +469,7 @@ void PHY_ext_cmd(phyMessageHeader_t *msg, instanceId_t phy_instance)
 
     if (!phy_intf_init_done || (msg->msgType != gPhyExtCmd))
     {
+        wcs_trace_rpmsg_phy(true, msg->msgType, (macToPlmeMessage_t *)msg, phy_instance, gPhyInvalidParameter_c);
         return;
     }
 
@@ -476,8 +489,10 @@ void PHY_ext_cmd(phyMessageHeader_t *msg, instanceId_t phy_instance)
 
         if (HAL_RpmsgSend((hal_rpmsg_handle_t)phyRpmsgHandle, (uint8_t *)msg, len) != kStatus_HAL_RpmsgSuccess)
         {
+             wcs_trace_rpmsg_phy(true, msg->msgType, (macToPlmeMessage_t *)msg, phy_instance, gPhyBusy_c);
              break;
         }
+        wcs_trace_rpmsg_phy(true, msg->msgType, (macToPlmeMessage_t *)msg, phy_instance, gPhyBusy_c);
     } while (0);
 
     PLATFORM_RemoteActiveRel();
